lib/puppet/provider/package/pip.rb

```diff
@@ -113,11 +113,23 @@ def update
   def lazy_pip(*args)
     pip *args
   rescue NoMethodError => e
-    if pathname = which(self.class.cmd)
-      self.class.commands :pip => pathname
-      pip *args
+    if self.class.cmd.class == Array
+      if pathname = which(self.class.cmd.first)
```

Comment:
Same assumption as above about the number of elements in the array.  If some distro comes up with yet another way to name "pip", someone might append the new name to the array and suddenly skip `pip-python` checks, for example.